packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -65,11 +72,244 @@ describe('Sayt Component', () => {
+    });
+
+    it('should hide SAYT if the event target is nowhere relevant', () => {
+      sayt.contains = stub().returns(false);
+      sayt.hideSayt = spy();
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.be.called;
+    });
+
+    it('should not SAYT if the event target is contained by SAYT', () => {
+      sayt.contains = stub().returns(true);
+      sayt.hideSayt = spy();
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.not.be.called;
+    });
+
+    it('should not hide SAYT if the event target is the provided search box', () => {
+      sayt.contains = stub().returns(false);
+      sayt.nodeInSearchBar = stub().returns(true);
+      sayt.hideSayt = spy();
+
+      sayt.processClick(event);
+
+      expect(sayt.contains).to.be.calledWith(node);
+      expect(sayt.nodeInSearchBar).to.be.calledWith(node);
+      expect(sayt.hideSayt).to.not.be.called;
+    });
+  });
+
+  describe('nodeInSearchBar()', () => {
+    it('should return true if given node is contained in the search bar', () => {
+      const searchbar = {
+        contains: stub().returns(true),
+      };
+      const querySelector = stub(document, 'querySelector').callsFake(() => searchbar);
+      sayt.searchbar = 'searchbar-id';
+
+      const result = sayt.nodeInSearchBar('node');
+
+      expect(querySelector).to.be.calledWith('#searchbar-id');
+      expect(searchbar.contains).to.be.calledWith('node');
+      expect(result).to.equal(true);
+    });
+
+    it('should return false if given node is not contained in the search bar', () => {
+      const searchbar = {
+        contains: stub().returns(false),
+      };
+      const querySelector = stub(document, 'querySelector').callsFake(() => searchbar);
+      sayt.searchbar = 'searchbar-id';
+
+      const result = sayt.nodeInSearchBar('node');
+
+      expect(querySelector).to.be.calledWith('#searchbar-id');
+      expect(searchbar.contains).to.be.calledWith('node');
+      expect(result).to.equal(false);
+    });
+
+    it('should return false if there is no search bar on the page', () => {
+      const querySelector = stub(document, 'querySelector').returns(null);
+
+      const result = sayt.nodeInSearchBar('node');
+
+      expect(result).to.be.false;
+    });
+
+    it('should not call document.querySelector() if this.searchbar is not set', () => {
+      sayt.searchbar = undefined;
+      const querySelector = stub(document, 'querySelector');
+
+      sayt.nodeInSearchBar('node');
+
+      expect(querySelector).to.not.be.called;
+    });
+  });
+
+  describe('processKeyPress()', () => {
+    it('should exist as a function', () => {
+      expect(sayt.processKeyPress).to.be.a('function');
+    });
+
+    it('should hide SAYT when pressing escape', () => {
+      const event = { key: "Escape" } as KeyboardEvent;
+      sayt.hideSayt = spy();
```

Comment:
Same thing about this being a stub. I won't flag the rest of the occurrences, but it applies to all of them as well.